mobilecoind/src/service.rs

```diff
@@ -1118,26 +1118,38 @@ impl<
 
     fn get_tx_status_as_sender_impl(
```

Comment:
The main change in this PR is to update this method to reflect the 3 new failure modes.